licenses/management/commands/publish.py

```diff
@@ -0,0 +1,172 @@
+from django_distill.distill import urls_to_distill
+from django_distill.errors import DistillError
+from django_distill.renderer import render_to_dir
+
+from cc_licenses.settings.base import TRANSLATION_REPOSITORY_DIRECTORY
+from licenses.utils import cleanup_current_branch_output, strip_list_whitespace
+
+
+def run_django_distill(obj: object):
+    """Outputs static files into the specified directory determined by settings.base.DISTILL_DIR
+    """
+    stdout = obj._quiet
+    output_dir = getattr(settings, "DISTILL_DIR", None)
+    if not os.path.isdir(settings.STATIC_ROOT):
+        e = "Static source directory does not exist, run collectstatic"
+        raise CommandError(e)
+    output_dir = os.path.abspath(os.path.expanduser(output_dir))
+    if os.path.isdir(output_dir):
+        rmtree(output_dir)
+        os.makedirs(output_dir)
+    else:
+        os.makedirs(output_dir)
+    try:
+        render_to_dir(output_dir, urls_to_distill, stdout)
+    except DistillError as err:
+        raise CommandError(str(err)) from err
+
+
+def check_if_build_to_push(branch: str):
+    """Navigates to a branch and checks the status of the branch
+
+    Returns:
+        True if files need to be committed
+        False if the working tree is clean
+    """
+    subprocess.run(
+        ["git", "checkout", f"{branch}"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+    )
+    status = (
+        subprocess.check_output(["git", "status"], cwd=TRANSLATION_REPOSITORY_DIRECTORY)
+        .decode()
+        .splitlines()
+    )
+    if "nothing to commit, working tree clean" in status:
+        return False
+    return True
+
+
+def git_on_branch_and_pull(branch: str):
+    """Commands to checkout and pull from a branch"""
+    subprocess.run(
+        ["git", "checkout", f"{branch}"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+    )
+    return subprocess.run(
+        ["git", "pull", "origin", f"{branch}"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+    )
+
+
+def git_commit_and_push(branch: str):
+    """Command to git checkout, commit, and push branch"""
+    subprocess.run(["git", "add", "build/"], cwd=TRANSLATION_REPOSITORY_DIRECTORY)
+    subprocess.run(
+        ["git", "commit", "-m", f"{branch}"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+    )
+    return subprocess.run(
+        ["git", "push", "origin", f"{branch}"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+    )
+
+
+def pull_translations_branches():
+    """Git pulls branches in cc-licenses-data to update local git registry"""
+    subprocess.run(["git", "checkout", "develop"], cwd=TRANSLATION_REPOSITORY_DIRECTORY)
+    return subprocess.run(["git", "pull"], cwd=TRANSLATION_REPOSITORY_DIRECTORY)
+
+
+def list_open_branches():
+    """List of open branches in cc-licenses-data repo
+    """
+    pull_translations_branches()
+    branches = (
+        subprocess.check_output(
+            ["git", "branch", "--list"], cwd=TRANSLATION_REPOSITORY_DIRECTORY
+        )
+        .decode()
+        .split("\n")
```

Comment:
Can we use ```.splitlines()``` here too? That way we won't get an empty string at the end if the output ended with ```\n```.

Reply:
done